### ISSUE TITLE
CPU OOM Finetuning on custom data

### ISSUE THREAD

**Issue by mikygit** [moojink/openvla-oft#59]
Hello,
 I searched in the Issues but no one seems to have OOM issues when finetuning.

I tried twicking the params but it does not help. Reducing the shuffle_buffer_size barely help.
It makes it difficult to use unless you happen to have tones of CPU mem.

The CPU mem consumption keeps growing and growing every round.
Am i the only one facing this pb?

Any help would be appreciated thx.

**Comment by Yifeng-Bai** [comment]
Hi，
When I was using distributed training on a single machine with four H100 GPUs, I also encountered this issue. The CPU memory would be exhausted after training for a period of time, and then the training process would be terminated by system Signal 9 (SIGKILL). Have you eventually resolved this issue?

**Comment by may210297** [comment]
> Hi， When I was using distributed training on a single machine with four H100 GPUs, I also encountered this issue. The CPU memory would be exhausted after training for a period of time, and then the training process would be terminated by system Signal 9 (SIGKILL). Have you eventually resolved this issue?

Same issue.

**Comment by qiushui0618** [comment]
This is due to an issue with TFDS dataloader. You can resolve this by setting `shuffle_buffer_size`. See https://github.com/openvla/openvla/issues/4 for details.
